javaslang/src/main/java/javaslang/collection/Vector.java

```diff
@@ -164,14 +164,21 @@ private Vector(BitMappedTrie<T> trie) {
         if (elements instanceof Vector) {
             return (Vector<T>) elements;
         } else {
-            BitMappedTrie<T> trie = BitMappedTrie.empty();
-            for (T element : elements) {
-                trie = trie.append(element);
-            }
-            return ofAll(trie);
+            final Seq<? extends T> seq = seq(elements);
+            return ofAll(seq, seq.size());
         }
     }
 
+    public static <T> Vector<T> ofAll(Collection<? extends T> elements) {
```

Comment:
Do we really need that extra method? Why use the `ofAll(Iterable)` method and check `instanceof Collection` in order to call `size()`? Would it really be significant slower? Other collections do not have that `ofAll` method.